include/util.h

```diff
@@ -6,6 +6,8 @@
 
 #include "log.h"
 
+#define MAX_FILEPATH 256
```

Comment:
Let's use [`limits.h`](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/limits.h.html) on POSIX. As a result rename to PATH_MAX.

 `#define PATH_MAX        4096	/* # chars in a path name including nul */`

Simply, 
```c
#ifdef __linux__ 
    #include <limits.h>
#else
#define PATH_MAX ...
#endif
```

Reply:
wont we have a problem complining on macOS using `limits.h`?

Reply:
No, that's where the `#else` takes action. But we can also do conditional compilation for POSIX systems via:

```c
#ifdef _POSIX_C_SOURCE
#include <limits.h>
#else
#define PATH_MAX ... // For non-POSIX systems (unlikely but let'ts be safe)
#endif
```

Please double check if `_POSIX_C_SOURCE` is the correct macro, I might remember something wrong.